### PR TITLE
fix: rebuild focused backend fixes

### PR DESCRIPTION
**Pull request by BaconDroid**
## Summary

Rebuilds the focused backend fixes from #292 on current `main`. Frontend Lua changes are intentionally excluded.

## Fix mapping

| Fix | Replacement commits | Rebuilds #292 commits |
| --- | --- | --- |
| SQLite chapter upserts respect the bind limit and refresh `lang`. | `d658128` | `a7de7c8` |
| Batch downloads handle numbered and unnumbered chapters without crossing the read boundary. | `21ff311`, `a22e992`, `d0b6d33`, `fe0a70d` | `9b5d612` |
| The non-default shared build retains its progress callback and result mapping. | `ba76c92` | `f64080e` |
| Manga refresh endpoints return failures to clients. | `f895d7a` | `afc2e57` |
| Remote covers remain available when no local poster resolves; conversion failures use server logging. | `9752bea`, `9fcacfc` | `579a3d9` |

## Validation

- `cargo test --workspace`
- `cargo fmt --manifest-path backend/Cargo.toml --all -- --check`
- `cargo check -p server`

Supersedes #292.

### CODE REVIEW

File: frontend/rakuyomi.koplugin/jobs/DownloadUnreadChapters.lua [hunk 1]
```diff
@@ -66,7 +66,7 @@ function DownloadUnreadChapters:start()
   end
 
   if response.type == 'ERROR' then
-    logger.error('could not create download job', response.message)
+    logger.err('could not create download job', response.message)
```
Comment (tachibana-shin) [inline frontend/rakuyomi.koplugin/jobs/DownloadUnreadChapters.lua]: logger.error is correct logger.err as alias not need

**Review by tachibana-shin — CHANGES_REQUESTED** [review]

**Comment by tachibana-shin** [comment]
Basically, Rust code is fine, but Lua has way too many issues

**Comment by BaconDroid** [comment]
Updated this PR to remove the Lua changes, as requested; it now contains only the Rust fixes.

If there are repository conventions or review expectations I missed, I would appreciate any pointers so I can align future contributions before opening a PR.

**Review by tachibana-shin — COMMENTED** [review]
Reviewed all changes against current `main` — nothing needs to be removed. Verification results:

**Confirmed real bugs being fixed:**
- `INSERT_FIELD_COUNT = 8` vs 12 actual columns: `CHUNK_SIZE = 32766 / 8 = 4095` → full chunks bind 49140 params > SQLite's 32766 limit, so chapter upserts fail for manga with >2730 chapters. The `lang = excluded.lang` addition also fixes stale languages on refresh.
- Non-default build (`--no-default-features --features sqlx`) currently fails to compile on `main` with 4 errors: E0382 `on_progress` move, E0599 `InvalidRuleList` (x2). Both fixes are correct.
- The batch-filter rewrite correctly fixes unnumbered-chapter boundary leaks; tests cover the edge cases well.

**One missing fix** — the non-default build still doesn't compile without this (verified on your branch):

```diff
--- a/backend/shared/src/source/lnreader/manifest.rs
+++ b/backend/shared/src/source/lnreader/manifest.rs
@@ -105,6 +105,8 @@ pub fn manifest_from_props(
         info: SourceInfo {
             id: props.id.clone(),
             lang: None,
             languages: None,
+            #[cfg(not(feature = "all"))]
+            content_rating: None,
             name: props.name.clone(),
             version: serde_json::Value::String(props.version.clone()),
             url: Some(props.site.clone()),
```

With that applied, both `cargo check -p shared` and `cargo check -p shared --no-default-features --features sqlx` pass, and all 278 lib tests pass.

**Comment by BaconDroid** [comment]
Fixed in 940b64d. I added the cfg-gated `content_rating` initializer in the LNReader manifest and verified:

- `cargo check -p shared`
- `cargo check -p shared --no-default-features --features sqlx`
- `cargo test -p shared --lib` (278 passed)

**Review by tachibana-shin — APPROVED** [review]
